control-plane/gateways/deployment_init_container.go

```diff
@@ -0,0 +1,183 @@
+	container := corev1.Container{
+		Name:  initContainerName,
+		Image: config.ImageConsulK8S,
+
+		Env: []corev1.EnvVar{
+			{
+				Name: "POD_NAME",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.name"},
+				},
+			},
+			{
+				Name: "POD_NAMESPACE",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.namespace"},
+				},
+			},
+			{
+				Name: "NODE_NAME",
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "spec.nodeName",
+					},
+				},
+			},
+			{
+				Name:  "CONSUL_ADDRESSES",
+				Value: config.ConsulConfig.Address,
+			},
+			{
+				Name:  "CONSUL_GRPC_PORT",
+				Value: strconv.Itoa(config.ConsulConfig.GRPCPort),
+			},
+			{
+				Name:  "CONSUL_HTTP_PORT",
+				Value: strconv.Itoa(config.ConsulConfig.HTTPPort),
+			},
+			{
+				Name:  "CONSUL_API_TIMEOUT",
+				Value: config.ConsulConfig.APITimeout.String(),
+			},
+			{
+				Name:  "CONSUL_NODE_NAME",
+				Value: "$(NODE_NAME)-virtual",
+			},
+		},
+		VolumeMounts: volMounts,
+		Command:      []string{"/bin/sh", "-ec", buf.String()},
+	}
+
+	if config.TLSEnabled {
+		container.Env = append(container.Env,
+			corev1.EnvVar{
+				Name:  "CONSUL_USE_TLS",
+				Value: "true",
+			},
+			corev1.EnvVar{
+				Name:  "CONSUL_CACERT_PEM",
+				Value: config.ConsulCACert,
+			},
+			corev1.EnvVar{
+				Name:  "CONSUL_TLS_SERVER_NAME",
+				Value: config.ConsulTLSServerName,
+			})
+	}
+
+	if config.AuthMethod != "" {
+		container.Env = append(container.Env,
+			corev1.EnvVar{
+				Name:  "CONSUL_LOGIN_AUTH_METHOD",
+				Value: config.AuthMethod,
+			},
+			corev1.EnvVar{
+				Name:  "CONSUL_LOGIN_BEARER_TOKEN_FILE",
+				Value: bearerTokenFile,
+			},
+			corev1.EnvVar{
+				Name:  "CONSUL_LOGIN_META",
+				Value: "pod=$(POD_NAMESPACE)/$(POD_NAME)",
+			})
+
+		if config.ConsulTenancyConfig.ConsulPartition != "" {
+			container.Env = append(container.Env, corev1.EnvVar{
+				Name:  "CONSUL_LOGIN_PARTITION",
+				Value: config.ConsulTenancyConfig.ConsulPartition,
+			})
+		}
+	}
+	container.Env = append(container.Env,
+		corev1.EnvVar{
+			Name:  "CONSUL_NAMESPACE",
+			Value: consulNamespace,
+		})
+
+	if config.ConsulTenancyConfig.ConsulPartition != "" {
+		container.Env = append(container.Env,
+			corev1.EnvVar{
+				Name:  "CONSUL_PARTITION",
+				Value: config.ConsulTenancyConfig.ConsulPartition,
+			})
+	}
+
+	return container, nil
+}
+
+// initContainerCommandTpl is the template for the command executed by
+// the init container.
+// TODO @GatewayManagement parametrize gateway kind.
+const initContainerCommandTpl = `
+consul-k8s-control-plane connect-init -pod-name=${POD_NAME} -pod-namespace=${POD_NAMESPACE} \
+	-gateway-kind="mesh-gateway" \
+  -log-json={{ .LogJSON }} \
+  {{- if .AuthMethod }}
+  -service-account-name="{{ .ServiceAccountName }}" \
+  {{- end }}
+  -service-name="{{ .ServiceName }}"
```

Comment:
#nit just for consistent indentation

```suggestion
consul-k8s-control-plane connect-init \
  -pod-name=${POD_NAME} \
  -pod-namespace=${POD_NAMESPACE} \
  -gateway-kind="mesh-gateway" \
  -log-json={{ .LogJSON }} \
  {{- if .AuthMethod }}
  -service-account-name="{{ .ServiceAccountName }}" \
  {{- end }}
  -service-name="{{ .ServiceName }}"
```